website/static/js/pages/file-page.js

```diff
@@ -4,4 +4,42 @@ var $osf = require('js/osfHelpers');
 var FileViewPage = require('js/filepage');
 var waterbutler = require('js/waterbutler');
 
+require('jquery-tagsinput');
+
 m.mount(document.getElementsByClassName('file-view-panels')[0], FileViewPage(window.contextVars));
+
+$(function() {
+    // Tag input
+    $('#fileTags').tagsInput({
+        width: '100%',
+        interactive: window.contextVars.currentUser.canEdit,
+        maxChars: 128,
+        onAddTag: function (tag) {
+            var url = '/api/v1/project/' + window.contextVars.node.id + '/osfstorage' + window.contextVars.file.path +
```

Comment:
I'm not a big fan of constructing URLs piecemeal in the template this way; it seems very fragile. Can you pass the URLs in from the view function instead? (in order to take advantage of `absolute_url_for`)